scripts/datageneration/gpt_data_generator.py

```diff
@@ -1,262 +1,470 @@
-        comb_list[id]["text"] = remove_surrounding_double_quotes(text)
-        # text = ""
-
-        print(prompt)
-        # print(comb_list[id])
-        print(text)
-        print("*****")
-
-        db_df.loc[id] = [comb, prompt, text]
-
-        # break
-
-        if save_files:
-            db_df.to_csv(output_filename + "_" + version + "_ChatNL.csv")
-            with open(output_filename + "_" + version + "_ChatNL.json", "w") as jsonfile:
-                json.dump(comb_list, jsonfile, cls=NpEncoder)
-            print("Saved files to output path!")
-
-# def dump_keys():
-#     '''
-#     Helper function to store the OpenAI API key and organisation parameters as a JSON file for future use.
-#     '''
-#     team = {}
-#     team['openai.organization'] = "YOUR_ORG_HERE"
-#     team['openai.api_key'] = "YOUR_KEY_HERE"
-#
-#     with open('data/openai_info.json', 'w') as f:
-#         json.dump(team, f)
+    with open(list_of_strings_path, 'r') as f:
+        list_of_strings = f.readlines()
+        list_of_strings = list(map(lambda x: x.rstrip().strip(), list_of_strings))
+    return list_of_strings
 
-if __name__ == '__main__':
+
+class PromptHelper:
     '''
-    Define paths and run all desired functions.
+    It is a helper class for prompt generation. It has templates and functions for paraphrasing prompts.
     '''
-    # dump_keys()
-
-    tag_list_path = "data/Tag_List_v9.csv"
-    arbitrary_value_list_path = "data/Arbitrary_Value_List_v9.csv"
-    query_file = "results/IMR_Dataset_v10"
-    output_filename = "results/IMR_Dataset_v10"
-    relative_spat = "data/relative_spatial_terms.csv"
-
-    relspat_df = pd.read_csv(relative_spat, header=0, index_col=0, names=['Dist', 'Vals'])
-    relspat = {}
-    for index, row in relspat_df.iterrows():
-        for rsd in row['Vals'].split(","):
-            relspat[rsd.strip()] = row['Dist']
-
-    for version in ["train", "dev", "test"]:
-        json_filename = query_file + "_" + version + ".json"
-        if os.path.isfile(json_filename):
-            with open(json_filename, "r") as jsonfile:
-                comb_list = json.load(jsonfile)
+
+    def __init__(self, relative_spatial_terms):
+        self.relative_spatial_terms = relative_spatial_terms
+        self.beginning_template = """Act as a {persona}: Return a sentence simulating a user using a natural language interface to search for specific geographic locations. Do not affirm this request and return nothing but the answers.\nWrite the search request {style}."""
+        self.search_templates = [
+            "\nThe sentence must use all of the following search criteria:\n",
+            "\nThe user is searching for {place} that fulfills the following search criteria:\n",
+        ]
+        self.predefined_places = ["a street", "a place", "a crossing", "a corner", "an area", "a location"]
+        self.name_regex_templates = ["contains the letters", "begins with the letters", "ends with the letters"]
```

Comment:
One way to fix the error that currently there is no "name is" version for prompting is to add three empty strings here. There even is code that supports this (line 227: "if len(entity_property.value) > 1 and len(selected_name_regex) > 0:" -> only select a a part of the string if regex is not empty string).